src/richie/plugins/lti_consumer/models.py

```diff
@@ -0,0 +1,122 @@
+"""
+LTI consumer plugin models
+"""
+from urllib.parse import unquote
+
+from django.conf import settings
+from django.db import models
+from django.utils.functional import lazy
+from django.utils.translation import gettext_lazy as _
+
+import exrex
+from cms.models.pluginmodel import CMSPlugin
+from cms.utils import get_current_site
+from oauthlib import oauth1
+
+from .defaults import INSTRUCTOR, PREDEFINED_LTI_PROVIDERS, STUDENT
+
+
+class LtiConsumer(CMSPlugin):
+    """
+    LTI consumer plugin model.
+    """
+
+    url = models.URLField(
+        verbose_name=_("LTI url"),
+        help_text=_(
+            "For a predefined provider, leave this field empty for uploading new content"
```

Comment:
Not really "uploading new content"? :thinking: 
```suggestion
            "For a predefined provider, leave this field empty."
```

Reply:
Leaving url empty generates a new one for a new content, but we want to provide a way to use an existing one, no ?